app/models/vim_performance_state.rb

```diff
@@ -48,31 +48,32 @@ def self.capture(obj)
     state = obj.vim_performance_states.find_by_timestamp(ts)
     return state unless state.nil?
 
-    state = obj.vim_performance_states.build
-    state.state_data ||= {}
-    state.timestamp = ts
-    state.capture_interval = 3600
-    state.assoc_ids = capture_assoc_ids(obj)
-    state.parent_host_id = capture_parent_host(obj)
-    state.parent_storage_id = capture_parent_storage(obj)
-    state.parent_ems_id = capture_parent_ems(obj)
-    state.parent_ems_cluster_id = capture_parent_cluster(obj)
-    # TODO: This is cpu_total_cores and needs to be renamed, but reports depend on the name :numvcpus
-    state.numvcpus = capture_cpu_total_cores(obj)
-    state.total_cpu = capture_total(obj, :cpu_speed)
-    state.total_mem = capture_total(obj, :memory)
-    state.reserve_cpu = capture_reserve(obj, :cpu_reserve)
-    state.reserve_mem = capture_reserve(obj, :memory_reserve)
-    state.vm_used_disk_storage = capture_vm_disk_storage(obj, :used_disk)
-    state.vm_allocated_disk_storage = capture_vm_disk_storage(obj, :allocated_disk)
-    state.tag_names = capture_tag_names(obj)
-    state.image_tag_names = capture_image_tag_names(obj)
-    state.host_sockets = capture_host_sockets(obj)
+    state = obj.vim_performance_states.build(:timestamp => ts)
+    state.capture
     state.save
 
     state
   end
 
+  def capture
+    self.state_data ||= {}
+    self.capture_interval = 3600
+    self.assoc_ids = VimPerformanceState.capture_assoc_ids(resource)
+    self.parent_host_id = VimPerformanceState.capture_parent_host(resource)
+    self.parent_storage_id = VimPerformanceState.capture_parent_storage(resource)
+    self.parent_ems_id = VimPerformanceState.capture_parent_ems(resource)
+    self.parent_ems_cluster_id = VimPerformanceState.capture_parent_cluster(resource)
+    capture_cpu_total_cores
+    self.total_cpu = VimPerformanceState.capture_total(resource, :cpu_speed)
+    self.total_mem = VimPerformanceState.capture_total(resource, :memory)
+    self.reserve_cpu = VimPerformanceState.capture_reserve(resource, :cpu_reserve)
+    self.reserve_mem = VimPerformanceState.capture_reserve(resource, :memory_reserve)
+    capture_vm_disk_storage
```

Comment:
This is now consistent.

Now the instance variable and the code that sets values for such instance variable are in one place. That is helpful when searching for it.

Also the interface is now a little more clean, as things that should have been private are private, and things that work with single VimPerformanceState are instance method.